packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -43,48 +54,79 @@ export function getNearestBlockIndex( elements, position, orientation ) {
 	const isRightToLeft = isRTL();
 
 	let candidateIndex;
+	let candidatePosition = 'after';
 	let candidateDistance;
 
 	elements.forEach( ( element, index ) => {
 		const rect = element.getBoundingClientRect();
-		const [ distance, edge ] = getDistanceToNearestEdge(
+
+		let [ distance, edge ] = getDistanceToNearestEdge(
 			position,
 			rect,
 			allowedEdges
 		);
+		// Prioritize the element if the point is inside of it.
+		if ( isPointContainedByRect( position, rect ) ) {
+			distance = 0;
+		}
```

Comment:
I don't think I fully understand what you mean here 🤔, I recorded a video and followed what you mentioned but didn't notice anything weird.

https://user-images.githubusercontent.com/7753001/193998823-105af8db-23ee-4142-b1d5-e091397f6568.mp4

Could you share more details on this?

Reply:
Here's a couple of videos. In this PR, as soon as my cursor is slightly over the image, the drop indicator is showing before the image, even though that's not the closest place I can drop to, between '2' and '3' is still closer:

https://user-images.githubusercontent.com/677833/194002843-d1fe5929-ef66-4b0f-b126-8edf6f144adf.mp4

In trunk, this is what happens:

https://user-images.githubusercontent.com/677833/194003300-ced2691c-5a3b-432c-9eb4-9110375ff250.mp4

The flickering of the drop indicator isn't ideal, but it always chooses the nearest drop location. I think that's important because:
- We don't know what layout the user might have, the boundaries between blocks might not be very clear.
- In the video, I made the paragraphs quite big, but in reality they're usually much smaller and harder to drop between.

Reply:
I see! I'll see what I can do.

Reply:
I discovered one issue when trying to fix this. Why are we using `blockElements` and getting them manually rather than using `getBlocks()`? I found out that `blockElements` will contain the default block appender as well. I'm not sure when we'll want that?

Reply:
`getBlocks()` is only block data, and the code needs the DOM element's bounding rect. It's certainly not ideal how it relies on the DOM particulars of blocks and the block list.

I know there's now `useBlockElement` (it didn't exist at the time of writing this code), so it may now be possible to get the elements for an entire block list using more of an editor API.

If we have to do `getElementById` or `querySelector` in a map, then it'd definitely be worth checking on performance. That feels potentially expensive.

Reply:
I think it should be pretty fast to do `document.getElementById`. Calling it 1000000 times in a loop only spends around 30ms on my machine.

Reply:
I ended up refactored it to `getDropTargetPosition` in cce8d8f.